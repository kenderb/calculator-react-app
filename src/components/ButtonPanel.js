import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const buttons = [
  { name: 'group1', values: ['AC', '+/-', '%', 'รท'] },
  { name: 'group2', values: ['7', '8', '9', 'X'] },
  { name: 'group3', values: ['4', '5', '6', '-'] },
  { name: 'group4', values: ['1', '2', '3', '+'] },
  { name: 'group5', values: ['0', '.', '='] },
];

const ButtonPanel = ({ clickHandler }) => (
  <div>
    {buttons.map(item => (
      <div key={item.name}>
        {item.values.map(itemButton => (
          <Button
            buttonName={itemButton}
            key={itemButton}
            onClickBtn={clickHandler}
          />
        ))}
      </div>
    ))}

  </div>
);

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
