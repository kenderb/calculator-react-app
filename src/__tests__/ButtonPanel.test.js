import { create } from 'react-test-renderer';
import ButtonPanel from '../components/ButtonPanel';

test('renders correctly', () => {
  const tree = create(<ButtonPanel clickHandler={() => 'hello'} />).toJSON();
  expect(tree).toMatchSnapshot();
});
