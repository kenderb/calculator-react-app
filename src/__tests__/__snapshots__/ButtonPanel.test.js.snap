// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-gsTCUz dYdny"
>
  <div>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="sc-dlfnbm kLtawK"
      color="true"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>
  <div>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="sc-dlfnbm kLtawK"
      color="true"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </div>
  <div>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="sc-dlfnbm kLtawK"
      color="true"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>
  <div>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="sc-dlfnbm kLtawK"
      color="true"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>
  <div>
    <button
      className="sc-dlfnbm fbeBMw"
      color="false"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="sc-dlfnbm fLGXYK"
      color="false"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="sc-dlfnbm kLtawK"
      color="true"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
